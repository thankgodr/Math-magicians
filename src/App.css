.calculator {
  width: 36rem;
}

.result {
  float: right;
  margin: auto 0;
  color: white;
}

.resultview {
  padding: 8px;
  background-color: #858694;
}

.btn_number {
  padding: 20px;
  background-color: #e0e0e0;
  border-bottom: #d1d1d1 3px solid;
  border-right: #d1d1d1 3px solid;
}

.function {
  padding: 20px;
  background-color: #f5913f;
  border-bottom: #d1d1d1 3px solid;
}
