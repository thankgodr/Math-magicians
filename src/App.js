import React from 'react';
import { Route, Routes } from 'react-router-dom';
import './App.css';
import NAVIGATIONPATH from './common/routepaths';
import CalculatorScreen from './components/CalculatorScreen';
import Header from './components/Header';
import Home from './components/Home';
import Quotes from './components/Quotes';

export default class App extends React.Component {
  #appclassName = 'App'

  render() {
    return (
      <div className={this.#appclassName}>
       <Header />
       <Routes>
          <Route path={NAVIGATIONPATH.HOME} element={<Home />} />
          <Route path={NAVIGATIONPATH.CALCULATOR} element={<CalculatorScreen/>} />
          <Route path={NAVIGATIONPATH.QUOTES} element={<Quotes />} />
       </Routes>

      </div>
    );
  }
}
