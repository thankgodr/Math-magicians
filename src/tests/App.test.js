import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import App from "../App";

describe("Testing snapshot of App", () => {
  test("Snapshot of App view", () => {
    const elem = renderer
      .create(
        <BrowserRouter>
          <App />
        </BrowserRouter>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
