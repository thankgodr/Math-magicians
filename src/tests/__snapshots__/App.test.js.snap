// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing snapshot of App Snapshot of App view 1`] = `
<div
  className="App"
>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light justify-content-between p-2"
  >
    <h1
      className="navbar-brand"
    >
      Magic Calculator
    </h1>
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/cal"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/quotes"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="m-4"
  >
    <h3>
      Welcome to our Page
    </h3>
    <p>
      Justo urna potenti urna massa dignissimnullam sollicitudinduis diampellentesque dignissim ipsumpraesent. Diampellentesque ridiculus suspendisse leo non dictumst pellentesque inceptos maecenas interdum suspendisse molestie. Imperdiet sempellentesque porta euismod mollisfusce tinciduntsed vivamus iaculis diam duis fringilla tortornulla maecenas muspraesent platea. Mus est efficitur arcucurabitur euismod leoetiam cubilia ac dui ex. Class tinciduntsed turpisetiam volutpatvestibulum quis magnased vulputate fermentum utvestibulum arcucurabitur ligulanunc. Neque eratnulla tinciduntsed torquent accumsanpellentesque tincidunt eu mauris arcu mattis aliquam
    </p>
    <p>
      Justo urna potenti urna massa dignissimnullam sollicitudinduis diampellentesque dignissim ipsumpraesent. Diampellentesque ridiculus suspendisse leo non dictumst pellentesque inceptos maecenas interdum suspendisse molestie. Imperdiet sempellentesque porta euismod mollisfusce tinciduntsed vivamus iaculis diam duis fringilla tortornulla maecenas muspraesent platea. Mus est efficitur arcucurabitur euismod leoetiam cubilia ac dui ex. Class tinciduntsed turpisetiam volutpatvestibulum quis magnased vulputate fermentum utvestibulum arcucurabitur ligulanunc. Neque eratnulla tinciduntsed torquent accumsanpellentesque tincidunt eu mauris arcu mattis aliquam
    </p>
  </div>
</div>
`;
